scripts/local_playback/countdown_clock_playback.exs

```diff
@@ -0,0 +1,100 @@
+# This script allows you to download a CSV file from Splunk containing logs of past sent messages
+# and play them back against a locally running instance of Signs UI.
+
+Mix.install([{:httpoison, "~> 1.0"}])
+
+if System.get_env("SIGN_UI_URL") == nil do
+  IO.puts("Please set the SIGN_UI_URL environment variable")
+  System.halt(1)
+end
+
+if System.get_env("SIGN_UI_API_KEY") == nil do
+  IO.puts("Please set the SIGN_UI_API_KEY environment variable")
+  System.halt(1)
+end
+
+defmodule CountdownClockPlayback do
+  use GenServer
```

Comment:
I haven't personally tried it yet, but I wonder if `Task` might be a good fit here. It would enable the code to be written as a single `Enum.each` with a `Process.sleep` for the delay, which might be slightly cleaner than the `Process.send_after` approach. The current approach is perfectly reasonable though, so consider this just a potentially interesting exploration.

Reply:
Hmm interesting, so would Task be used to encapsulate each http request? Usually `Task` would be used to take synchronous code and make it concurrent, but here we're essentially doing the opposite. Another reason I went with the GenServer approach was to have a little more flexibility in accessing the list since I'm looking ahead at the next message to calculate the delay. Using `Enum.each`, how would you suggest determining how much to sleep the process by?

Reply:
I was envisioning using a single `Task` for the whole simulation, similar to how you might use an `async` function in JS to perform a long-running sequence of calls. But yeah, getting the timing information makes it slightly more complicated, you'd need to use e.g. `Enum.chunk_by` to get each consecutive pair first.